stack_validation.go

```diff
@@ -0,0 +1,44 @@
+package lifecycle
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/buildpacks/imgutil"
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/cmd"
+	"github.com/buildpacks/lifecycle/platform"
+)
+
+func ValidateStack(stackMD platform.StackMetadata, runImage imgutil.Image) error {
+	buildStackID, err := getBuildStack(stackMD)
+	if err != nil {
+		return err
+	}
+
+	runStackID, err := runImage.Label(platform.StackIDLabel)
+	if err != nil {
+		return errors.Wrap(err, "get run image label")
+	}
+	if runStackID == "" {
+		return errors.New("get run image label: io.buildpacks.stack.id")
+	}
```

Comment:
I would put these lines under a `getRunStack` function similar to what you did for the `buildStackID`.